src/proxy/src/main/java/ee/ria/xroad/proxy/serverproxy/ServerRestMessageProcessor.java

```diff
@@ -232,6 +234,19 @@ private void updateOpMonitoringDataByRequest() {
                         + decoder.getAttachmentsByteCount());
             }
         }
+
+        if (requestMessage.getRest() != null) {
+            // Set request data into the opmon object
+            opMonitoringData.setServiceId(requestServiceId);
+            opMonitoringData.setRequestAttachmentCount(decoder.getAttachmentCount());
+            opMonitoringData.setRequestRestSize(requestMessage.getRest().getMessageBytes().length);
+
+            if (decoder.getAttachmentCount() > 0) {
+                opMonitoringData.setRequestMimeSize(requestMessage.getRest().getMessageBytes().length
+                        + decoder.getAttachmentsByteCount());
```

Comment:
The decoder does not update the attachment byte count for rest bodies, so the size is wrong.

Reply:
Decoder now updates the attachment byte count